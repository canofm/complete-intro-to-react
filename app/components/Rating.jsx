// @flow

import React from 'react';
import Spinner from './Spinner';

const RatingValue = (props: { value: string }) => <h3>{props.value}</h3>;

const Rating = (props: { value?: string }) =>
  props.value ? <RatingValue {...props} /> : <Spinner />;

Rating.defaultProps = {
  value: ''
};

export default Rating;
