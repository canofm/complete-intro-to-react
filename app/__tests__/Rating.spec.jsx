import React from 'react';
import { shallow } from 'enzyme';
import Rating, { RatingValue } from '../components/Rating';
import Spinner from '../components/Spinner';

describe('Rating component', () => {
  describe('by default', () => {
    const component = shallow(<Rating />);
    it('should renders correctly', () => {
      expect(component).toMatchSnapshot();
    });
    it('should contains Spinner component', () => {
      component.contains(<Spinner />);
    });
  });
  describe('when it gets value', () => {
    const value = '5.5';
    const component = shallow(<Rating value={value} />);
    it('should renders correctly', () => {
      expect(component).toMatchSnapshot();
    });
    it('should contains RatingValue component', () => {
      component.contains(<RatingValue value={value} />);
    });
  });
});
