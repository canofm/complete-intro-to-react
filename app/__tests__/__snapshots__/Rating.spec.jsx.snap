// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component by default should renders correctly 1`] = `<Spinner />`;

exports[`Rating component when it gets value should renders correctly 1`] = `
<RatingValue
  value="5.5"
/>
`;
